packages/react-dom/src/__tests__/ReactDOMEventPropagation-test.js

```diff
@@ -16,12 +16,12 @@ describe('ReactDOMEventListener', () => {
   beforeEach(() => {
     window.TextEvent = function() {};
     jest.resetModules();
-    React = require('react');
     jest.isolateModules(() => {
-      OuterReactDOM = require('react-dom');
+      InnerReactDOM = require('react-dom');
     });
     jest.isolateModules(() => {
-      InnerReactDOM = require('react-dom');
+      React = require('react');
+      OuterReactDOM = require('react-dom');
```

Comment:
Fixes https://app.circleci.com/pipelines/github/facebook/react/21294/workflows/b23b2fbb-9d67-447c-b395-21591b60ebf1/jobs/411166.

Probably required due to https://github.com/facebook/jest/pull/10963. Though it's unclear why this test passed without this change using `scripts/jest/config.source-www.js` but failed using `scripts/jest/config.build.js`.

@SimenB Any idea if this is intended?

Reply:
Sounds odd that changing config impacts module loading. My only guess is https://github.com/facebook/react/blob/c1220ebdde506de91c8b9693b5cb67ac710c8c89/scripts/jest/config.build.js#L38-L46 makes it resolve to differing modules somehow?

Reply:
The modules were all coming from the mapped paths. 
To be clear: It makes sense that the used `React` needs to be the same one that's used by `OuterReactDOM` and therefore needs to be in the same `isolatedModules` callback. What's odd is, that this isn't the case with `scripts/jest/config.source-www.js`.

So ideally we'd have two isolated copies of `react-dom` that somehow share the same `react` copy. This no longer seems possible with Jest 27

Reply:
Seems you've worked around this, but a solution might be a new API which lets you selectively evict specific modules from the cache? `jest.resetModule('react-dom')` or something?

Reply:
I haven't thought about this for a while but if I remember correctly, I came to the conclusiong that the old code should never have worked to begin with. Not sure if it's worth revisiting since this testing pattern is very specific.

Reply:
That sounds very reasonable 👍 It does sound like a weird edge case, but happy to revisit it if proves to actually guard against some regression 🙂